### ISSUE TITLE
cli: infer run number where possible

### ISSUE THREAD

**Issue by oliver-sanders** [cylc/cylc-flow#3895]
Attempt after ~#3889~, #4230

The default behaviour of `cylc install` will be to install workflows into an automatically named run (i.e. `run<N>`).

This means that the CLI will become a little cumbersome e.g.

```bash
# before
$ cylc stop my-flow
# after
$ cylc stop my-flow/run1
```

To combat this we should make the relevant Cylc subcommands (those with a REG argument) run-dir aware. That is to say that if the REG provided is not a workflow (denoted by the presence of a `.service` directory) but does contain automatically named runs (denoted by the presence of a `runN` symlink), then default to whatever run `runN` is pointing at.

This change returns the CLI REG argument to the way it was before for most cases whilst providing the ability to control named runs individually where required.

**Comment by MetRonnie** [comment]
Just to emphasise the last point (more of a note to self really): named runs (with `cylc install foo --run-name=bar`) should not be inferred (i.e. `cylc play foo` should not do anything, you would have to explicitly do `cylc play foo/bar`)

**Comment by MetRonnie** [comment]
Will attempt after #4180

**Comment by oliver-sanders** [comment]
Bumped the centralisation of workflow argument parsing into a separate issue to make this one a little clearer - https://github.com/cylc/cylc-flow/issues/4230
